modules/account-lib/src/coin/dot/walletInitializationBuilder.ts

```diff
@@ -1,26 +1,139 @@
+   *
+   * @see https://wiki.polkadot.network/docs/learn-proxies#proxy-types
+   */
+  type(proxyType: proxyType): this {
+    this._proxyType = proxyType;
+    return this;
+  }
+
+  /**
+   * The number of blocks that an announcement must be in place for.
+   * before the corresponding call may be dispatched.
+   * If zero, then no announcement is needed.
+   *
+   * @param {string} delay
+   * @returns {AddProxyBuilder} This transfer builder.
+   *
+   * @see https://wiki.polkadot.network/docs/learn-proxies#time-delayed-proxies
+   */
+  delay(delay: string): this {
+    this.validateValue(new BigNumber(parseInt(delay, 10)));
+    this._delay = delay;
+    return this;
+  }
+
   /** @inheritdoc */
-  protected async buildImplementation(): Promise<Transaction> {
-    throw new NotImplementedError('method not implemented');
+  validateRawTransaction(rawTransaction: string): void {
+    super.validateRawTransaction(rawTransaction);
+    const decodedTxn = decode(rawTransaction, {
+      metadataRpc: this._metadataRpc,
+      registry: this._registry,
+    });
+    if (decodedTxn.method?.name === MethodNames.AddProxy) {
+      const txMethod = decodedTxn.method.args as unknown as AddProxyArgs;
+      const delegate = txMethod.delegate;
+      const proxyType = txMethod.proxyType;
+      const delay = txMethod.delay;
+      const validationResult = WalletInitializationSchema.validate({ delegate, proxyType, delay });
+      if (validationResult.error) {
+        throw new InvalidTransactionError(`Transaction validation failed: ${validationResult.error.message}`);
+      }
+    }
   }
 
   /** @inheritdoc */
-  protected fromImplementation(rawTransaction: any): Transaction {
-    throw new NotImplementedError('method not implemented');
+  protected fromImplementation(rawTransaction: string): Transaction {
+    const tx = super.fromImplementation(rawTransaction);
+    if (this._method?.name === MethodNames.AddProxy) {
+      const txMethod = this._method.args as AddProxyArgs;
+      this.owner({ address: txMethod.delegate });
+      this.type(txMethod.proxyType);
+      this.delay(new BigNumber(txMethod.delay).toString());
+    } else {
+      throw new InvalidTransactionError(`Invalid Transaction Type: ${this._method?.name}. Expected addProxy`);
+    }
+    return tx;
   }
 
   /** @inheritdoc */
-  protected signImplementation(key: BaseKey): Transaction {
-    throw new NotImplementedError('method not implemented');
+  validateTransaction(_: Transaction): void {
+    super.validateTransaction(_);
+    this.validateFields(this._delegate, this._proxyType, this._delay);
+  }
+
+  private validateFields(delegate: string, proxyType: string, delay: string): void {
+    const validationResult = WalletInitializationSchema.validate({
+      delegate,
+      proxyType,
+      delay,
+    });
+
+    if (validationResult.error) {
+      throw new InvalidTransactionError(`AddProxy Transaction validation failed: ${validationResult.error.message}`);
```

Comment:
@argjv Not sure if here we need to continue with name pattern standard and replace AddProxy for WalletInit on exception messages.